src/main/java/gyro/google/compute/AbstractInstanceGroupManagerResource.java

```diff
@@ -0,0 +1,446 @@
+        List<InstanceGroupManagerVersion> versions = model.getVersions();
+
+        if (versions != null && !versions.isEmpty()) {
+            diffableInstanceGroupManagerVersion = versions
+                .stream()
+                .map(e -> {
+                    ComputeInstanceGroupManagerVersion computeVersion = newSubresource(
+                        ComputeInstanceGroupManagerVersion.class);
+                    computeVersion.copyFrom(e);
+                    return computeVersion;
+                })
+                .collect(Collectors.toList());
+        }
+        setVersion(diffableInstanceGroupManagerVersion);
+        setCurrentActions(Optional.ofNullable(model.getCurrentActions())
+            .map(e -> {
+                ComputeInstanceGroupManagerActionsSummary instanceGroupManagerActionsSummary = newSubresource(
+                    ComputeInstanceGroupManagerActionsSummary.class);
+                instanceGroupManagerActionsSummary.copyFrom(e);
+                return instanceGroupManagerActionsSummary;
+            })
+            .orElse(null));
+        setInstanceGroupLink(model.getInstanceGroup());
+        setSelfLink(model.getSelfLink());
+        setStatus(Optional.ofNullable(model.getStatus())
+            .map(e -> {
+                ComputeInstanceGroupManagerStatus computeInstanceGroupManagerStatus = newSubresource(
+                    ComputeInstanceGroupManagerStatus.class);
+                computeInstanceGroupManagerStatus.copyFrom(e);
+                return computeInstanceGroupManagerStatus;
+            })
+            .orElse(null));
+    }
+
+    protected InstanceGroupManager createInstanceGroupManager() {
+        InstanceGroupManager instanceGroupManager = new InstanceGroupManager();
+        instanceGroupManager.setBaseInstanceName(getBaseInstanceName());
+        instanceGroupManager.setName(getName());
+        instanceGroupManager.setTargetSize(getTargetSize());
+        instanceGroupManager.setAutoHealingPolicies(getAutoHealingPolicy()
+            .stream()
+            .map(ComputeInstanceGroupManagerAutoHealingPolicy::copyTo)
+            .collect(Collectors.toList()));
+        instanceGroupManager.setDescription(getDescription());
+        instanceGroupManager.setFingerprint(getFingerprint());
+        Optional.ofNullable(getInstanceTemplate())
+            .map(InstanceTemplateResource::getSelfLink)
+            .ifPresent(instanceGroupManager::setInstanceTemplate);
+        instanceGroupManager.setNamedPorts(getNamedPort().stream()
+            .map(ComputeNamedPort::copyTo)
+            .collect(Collectors.toList()));
+        // TODO: https://github.com/perfectsense/gyro-google-provider/issues/79
+        //        instanceGroupManager.setTargetPools(getTargetPools()
+        //            .stream()
+        //            .map(AbstractTargetPoolResource::getSelfLink)
+        //            .collect(Collectors.toList()));
```

Comment:
Should be removed and be done as a separate PR when target pool is available

Reply:
Yeah, it should be handled in a separate PR, but better to leave the comment as it will be helpful when implementing it.